File: LC-Python-Solution/3. Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:

        cSet = set()
        l = 0
        res = 0

        for i in range(len(s)):
            while s[i] in cSet:
                cSet.remove(s[l])
                l+=1
            
            cSet.add(s[i])
            res = max(res,i-l+1)
        
        return res









        

File: LC_JS-SOLUTIONS/3. Longest Substring Without Repeating Characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
  let cSet = new Set();
  let l = 0;
  let res = 0;

  for (let i = 0; i < s.length; i++) {
    while (cSet.has(s[i])) {
      cSet.delete(s[l]);
      l += 1;
    }
    cSet.add(s[i]);
    res = Math.max(res, i - l + 1);
  }

  return res;
};
